### PR TITLE
fix: collisions with other wallet providers

### LINKED ISSUE #160: [bug]: Filsnap adapter fails to connect when multiple wallet providers are installed
**Describe the bug**
Filsnap does not connect to MetaMask when multiple wallet browser extensions are installed. This seems to be due to the snap communicating with the wrong provider on the `window.ethereum` object. Might be related to missing support for [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963), could help to look at the [mpid](https://github.com/wevm/mipd) package for identifying the right provider.

**To Reproduce**
1. Install other wallet browser extension next to MetaMask (e.g. Rainbow, OKX Wallet)
2. Try to connect FilSnap to MetaMask
3. Connection fails with the error that MetaMask does not support snaps
4. Disable the other browerser extensions and connect again, it works

**Expected behavior**
FilSnap should connect to MetaMask even when multiple wallet providers are installed.

### PR DESCRIPTION
**Pull request by meandavejustice**
- fixes #160
- check for other existing providers which may cause conflict

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] Refactor (non-breaking change that updates existing functionality)

### CODE REVIEW

File: packages/adapter/test/adapter.spec.ts [hunk 1]
```diff
@@ -51,6 +51,29 @@ fixture.test(
...
+fixture = createFixture({
+  isolated: true,
+  download: {
+    flask: true,
+  },
+  snap: {
+    id: SNAP_ID,
+    version: SNAP_VERSION,
+  },
+  // Rainbow wallet extension id, once https://github.com/hugomrdias/filsnap-testing/pull/30 lands this can be uncommented
```
Comment (meandavejustice) [inline packages/adapter/test/adapter.spec.ts:63]: Add this back once https://github.com/hugomrdias/filsnap-testing/pull/30 is complete

**Comment by meandavejustice** [comment]
Getting some peculiar errors in this pr that I'm not seeing on master.

When running `pnpm test`:
After playwright's `webServer` is started on port `8081` in packages/snap,  getting logs for 404s on `http://localhost:8081` and `http://localhost:8081/index.html` until the timeout is triggered.

No tests are run as it seems to be waiting on the server to start.

I can confirm that the server _is_ running by fetching a file from that directory: `curl http://localhost:8081/snap.manifest.json`.

@hugomrdias Hoping you can help me out a bit here, I'm not understanding what/where is checking `localhost:8081`. Is this a playwright thing? This feels like a thing you'll probably know the answer to since you set these tests up.

**Comment by meandavejustice** [comment]
Digging in a bit more on this webserver failure, after updating packages in https://github.com/hugomrdias/filsnap-testing/pull/32 and updating some packages locally to get rid of the peer dependency issue, I'm seeing an actual error message:

```
j@Davids-MacBook-Air ~/Code/filsnap % pnpm test              

> root@ test /Users/dj/Code/filsnap
> pnpm -r --if-present run test

Scope: 5 of 6 workspace projects
packages/snap test$ playwright test
│ [WebServer] Analyzing[WebServer] 
│ [WebServer] [!] Error: Cannot find module '/Users/dj/Code/filsnap/node_modules/.pnpm/@metamask+key-tree@9.0.0/node_modules/@metamask/key-tree/dist/esm/BIP44Node' imported from /Users/dj/Code/f…
│ file:///Users/dj/Code/filsnap/node_modules/.pnpm/@metamask+key-tree@9.0.0/node_modules/@metamask/key-tree/dist/esm/BIP44Node
│     at finalizeResolution (node:internal/modules/esm/resolve:264:11)
│     at moduleResolve (node:internal/modules/esm/resolve:917:10)
│     at defaultResolve (node:internal/modules/esm/resolve:1130:11)
│     at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12)
│     at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25)
│     at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38)
│     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39)
│     at link (node:internal/modules/esm/module_job:84:36)
│ [WebServer] 
│ ❌ [build] exited with exit code 1.
│ [WebServer] ❌ 1 script failed.
│ Error: Process from config.webServer was not able to start. Exit code: 1
└─ Failed in 1s at /Users/dj/Code/filsnap/packages/snap
/Users/dj/Code/filsnap/packages/snap:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  filsnap@1.0.1 test: `playwright test`
Exit status 1
 ELIFECYCLE  Test failed. See above for more details.
```

There is something bizarre happening when importing this line in `snap/src/account.ts`

``` js
import { getBIP44AddressKeyDeriver } from '@metamask/key-tree'
```

I've verified locally that these files do exist in `node_modules/.pnpm/ {ETC ETC}`. 🤔 So not sure what could be causing this issue. 

@hugomrdias I did notice that you had run into previous issues with this library's imports as referenced here https://github.com/MetaMask/key-tree/issues/144, but it looks like this issue was resolved in `9.0.0`. I know tests have been failing for a while, perhaps this is still an issue in `@metamask/key-tree`

**Comment by hugomrdias** [comment]
its too deep i tried fixing this on our side, but can't find a quick fix

**Comment by meandavejustice** [comment]
@hugomrdias I see, we are using the newer style imports with ts and esm node. 

We should be able to just use the same module rules they are using in our tsconfig 

```
"module": "CommonJS",
"moduleResolution": "node",
```

and switch away from using esm at the package.json level.

That way we still get to use import statements in all of the actual codebase, but use requires in the rollup.js config, and we shouldnt have any issues with how dependencies are doing their packaging/importing.

**Comment by meandavejustice** [comment]
@hugomrdias Scratch that last comment. These test failures are happening on the master branch, and out of scope of this pr. Those test should be fixed elsewhere. I cleaned up the commit history to reflect the actual changes here.

**Comment by hugomrdias** [comment]
on main ? https://github.com/filecoin-project/filsnap/actions/workflows/manual.yml its passing

**Comment by hugomrdias** [comment]
metamask broke its own rollup plugin (kinda it now support proper esm but key-tree is not package right for esm) , and the cli serve cmd also doesnt work anymore it defaults to a full webpack build and serve, and they updated most snap packages to a proper esm packaging but forget key-tree....

**Comment by hugomrdias** [comment]
check this PR https://github.com/filecoin-project/filsnap/pull/163

**Comment by meandavejustice** [comment]
https://github.com/filecoin-project/filsnap/pull/163 accomplishes the updates and fixes weirdness around metamask packaging. Will rebase and fix up any remaining issues here once that lands :)

**Review by hugomrdias — APPROVED** [review]
LGTM

**Pull request merged**